src/main/java/com/hms_networks/americas/sc/extensions/datapoint/DataQuality.java

```diff
@@ -46,13 +104,47 @@ public static DataQuality fromRawDataQuality(int rawDataQuality) {
     return correspondingDataQualityObject;
   }
 
+  /**
+   * Gets the {@link DataQuality} object corresponding to the specified OPCUA quality value, or
+   * {@link DataQuality#UNCERTAIN} if the OPCUA quality value is unknown.
+   *
+   * <p>OPCUA quality values are 16-bit values with the following layout:
+   *
+   * <pre>
+   *   Bits 15-8: Vendor specific
+   *   Bits 7-6: Major quality (0-bad, 1-uncertain, 3-good)
+   *   Bits 5-2: Sub status code
+   *   Bits 1-0: Limit status
+   * </pre>
+   *
+   * <p>This method extracts the major quality bits and returns the corresponding {@link
+   * DataQuality} object.
+   *
+   * @param opcuaQuality OPCUA quality value
+   * @return corresponding {@link DataQuality} object
+   * @since 1.1.0
+   */
+  public static DataQuality fromOpcuaQuality(int opcuaQuality) {
+    int majorQuality = (opcuaQuality >> 6) & 0x03;
+    switch (majorQuality) {
+      case 3:
+        return DataQuality.GOOD;
+      case 0:
+        return DataQuality.BAD;
+      case 1:
+      default:
+        return DataQuality.UNCERTAIN;
+    }
```

Comment:
```suggestion
    final int qualityMask = 0x03;
    final int qualityPos = 6;
    final int qualityGood = 3;
    final int qualityBad = 0;
    final int qualityUncertain = 1;
    int majorQuality = (opcuaQuality >> qualityPos) & qualityMask;
    switch (majorQuality) {
      case qualityGood:
        return DataQuality.GOOD;
      case qualityBad:
        return DataQuality.BAD;
      case qualityUncertain:
      default:
        return DataQuality.UNCERTAIN;
    }
```

The constants dont have to live in the method here but i think they add value to explain what is going on even though the comments document the values.

Reply:
does 2 mean anything?

Reply:
> does 2 mean anything?

No
![image](https://github.com/hms-networks/sc-ewon-flexy-extensions-lib/assets/51409174/7a3830d3-9ea9-4168-8bb8-4a23e8835d8f)

Reply:
> The constants dont have to live in the method here but i think they add value to explain what is going on even though the comments document the values.

Made changes. I used existing constants for the 0/1/3 values and added in-method constants for 6 and 0x03.

Personally, though, I feel that these constants are not necessary here and ultimately make the code more difficult to follow because they introduce the need to jump around the code. The method Javadoc already thoroughly explains the intent and function of the method and identifies which bits correspond to each component of the quality value.